### DISCUSSION TITLE
Problem with dataset from xarray with filter ":TCDC:entire atmosphere"

### DISCUSSION THREAD

**Discussion by kdeblois** [blaylockbk/Herbie#381] [Q&A] [answered]
When I run the xarray command with fxx > 0 I get the following error: Note: Returning a list of [2] xarray.Datasets because cfgrib opened with multiple hypercubes. Running the command for a single level (e.g. "TCDC:1000 mb") works fine.
Any advice?

**Comment by aniruddhaadak80** [comment, marked as answer]
Hi @kdeblois ,

To resolve the `xarray` issue with multiple datasets when using `cfgrib` and filters like `fxx > 0`, try the following:

1. **Use Specific Filters**: Filter by a specific variable or level (e.g., `"TCDC:1000 mb"`), which avoids ambiguity and ensures only one hypercube is processed.

2. **Open Datasets Individually**: Use `xarray.open_mfdataset()` to load multiple datasets if needed, allowing you to merge them after loading.

3. **Apply Backend Filtering**: Specify `backend_kwargs={'filter_by_keys': {'typeOfLevel': 'surface'}}` in `open_dataset()` to narrow down the hypercubes being opened.

4. **Check Available Keys**: Call `.keys()` on the dataset to inspect variable levels, helping identify any discrepancies in your data.

These steps should help you work around the issue of receiving multiple datasets in your analysis.

Thank you .

**Reply by kdeblois** [reply]
Thanks so much!
